src/node_api.cc

```diff
@@ -1952,14 +1986,12 @@ napi_status napi_get_value_int32(napi_env env,
 
   if (val->IsInt32()) {
     *result = val.As<v8::Int32>()->Value();
-    return napi_clear_last_error(env);
-  }
-
-  RETURN_STATUS_IF_FALSE(env, val->IsNumber(), napi_number_expected);
+  } else {
+    RETURN_STATUS_IF_FALSE(env, val->IsNumber(), napi_number_expected);
 
-  v8::Isolate* isolate = env->isolate;
-  v8::Local<v8::Context> context = isolate->GetCurrentContext();
-  *result = val->Int32Value(context).FromJust();
+    v8::Local<v8::Context> context;  // empty context
+    *result = val->Int32Value(context).FromJust();
```

Comment:
Might be good to document why we are passing an empty context. A link to https://github.com/nodejs/node/issues/14379 shall suffice.